internal/product/accessor.go

```diff
@@ -0,0 +1,93 @@
+package product
+
+import (
+	"context"
+	"fmt"
+	"kg/procurement/internal/common/database"
+)
+
+const (
+	getProductsByVendorQuery = `
+		SELECT 
+			p.id, 
+			p.product_category_id, 
+			p.uom_id, 
+			p.income_tax_id, 
+			p.product_type_id, 
+			p.name, 
+			p.description, 
+			p.modified_date, 
+			p.modified_by 
+		FROM 
+			product p
+		JOIN 
+			product_vendor pv ON pv.product_id = p.id
+		WHERE 
+			pv.vendor_id = $1
+	`
```

Comment:
Why do you make the query variable global?

Reply:
it's not global per se, since it's lowercased
I simply did that to prevent the logic part of the code being too clunky